glean-core/ios/metrics.yaml

```diff
@@ -0,0 +1,238 @@
+      - 1497894
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1512938#c3
+    notification_emails:
+      - telemetry-client-dev@mozilla.com
+    expires: never
+
+  client_id:
+    type: uuid
+    description:
+      A UUID uniquely identifying the client.
+    send_in_pings:
+      - glean_client_info
+    lifetime: user
+    bugs:
+      - 1497894
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1512938#c3
+    notification_emails:
+      - telemetry-client-dev@mozilla.com
+    expires: never
+
+  app_build:
+    type: string
+    lifetime: application
+    send_in_pings:
+      - glean_client_info
+    description: |
+      The build identifier generated by the CI system (e.g. "1234/A").
+    bugs:
+      - 1508305
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1512938#c3
+    notification_emails:
+     - telemetry-client-dev@mozilla.com
+    expires: never
+
+  app_display_version:
+   type: string
+   lifetime: application
+   send_in_pings:
+     - glean_client_info
+   description: |
+     The user visible version string (e.g. "1.0.3").
+   bugs:
+     - 1508305
+   data_reviews:
+     - https://bugzilla.mozilla.org/show_bug.cgi?id=1508305#c9
+   notification_emails:
+     - telemetry-client-dev@mozilla.com
+   expires: never
+
+  app_channel:
+    type: string
+    lifetime: application
+    send_in_pings:
+     - glean_client_info
+    description: |
+     The channel the application is being distributed on.
+    bugs:
+      - 1520741
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1520741#c18
+    notification_emails:
+      - telemetry-client-dev@mozilla.com
+    expires: never
+
+  first_run_date:
+    type: datetime
+    lifetime: user
+    send_in_pings:
+      - glean_client_info
+    time_unit: day
+    description: |
+      The date of the first run of the application.
+    bugs:
+      - 1525045
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1525045#c18
+    notification_emails:
+      - telemetry-client-dev@mozilla.com
+    expires: never
+
+glean.error:
+ invalid_value:
+   type: labeled_counter
+   description:
+     Counts the number of times a metric was set to an invalid value.
+     The labels are the `category.name` identifier of the metric.
+   bugs:
+     - 1499761
+   data_reviews:
+     - https://bugzilla.mozilla.org/show_bug.cgi?id=1499761#c5
+   notification_emails:
+     - telemetry-client-dev@mozilla.com
+   expires: never
+   send_in_pings:
+     - all_pings
+
+ invalid_label:
+   type: labeled_counter
+   description:
+     Counts the number of times a metric was set with an invalid label.
+     The labels are the `category.name` identifier of the metric.
+   bugs:
+     - 1499761
+   data_reviews:
+     - https://bugzilla.mozilla.org/show_bug.cgi?id=1499761#c5
+   notification_emails:
+     - telemetry-client-dev@mozilla.com
+   expires: never
+   send_in_pings:
+     - all_pings
```

Comment:
Can we share the metrics.yaml and pings.yaml files between android and iOS?

Reply:
We might be, yes. Though that will require moving the files and linking them back (as both generators expect the files to be in the platform-specific project root).

Plus not all metrics line up, e.g. `device_manufacturer` will be a pretty boring metric on iOS, and `android_sdk_version` definitely isn't helpful on iOs.

Reply:
True. However, `device_manufacturer` is mandatory (required by our schema definition) so it must be there. `android_sdk_version` is optional, as it's Android specific.

Nothing prevents us from having multiple metrics.yaml files: the parsers can digest more than one registry file and will collapse things together :-) Regardless of the solution, it would be ok for me to have this as a follow-up, though.

Reply:
Good to know about `device_manufacturer` (and we should double-check when we implement the core metrics to have all required fields).

Yes, the Glean-internal parser should be easily able to pick up multiple files (we can specify them).
For users of course it should default to a root-level `metrics.yaml`.

definitely a follow-up to merge those files